build-tools/packages/build-tools/src/test/typeTestGenerator.test.ts

```diff
@@ -0,0 +1,243 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import * as utils from "../type-test-generator/typeTestUtils";
+import { readJsonSync } from "fs-extra";
+import * as fs from "fs";
+import * as path from "path";
+import { PackageJson } from "../common/npmPackage";
+import { IExtractorConfigPrepareOptions } from "@microsoft/api-extractor";
+
+/**
+ * Unit tests for the abstracted functions in typeTestUtils.
+ */
+describe("typeTestUtils", () => {
+	const packageJsonPath = path.join(__dirname, "mockPackage.json");
+	const packageObject: PackageJson = readJsonSync(packageJsonPath);
+	const previousPackageName = `${packageObject.name}-previous`;
+
+	describe("Test ensureDevDependencyExists", () => {
+		it("Should not throw an error if dev dependency exists", () => {
+			utils.ensureDevDependencyExists(packageObject, "dependency1");
+		});
+
+		it("Should throw an error if dev dependency does not exist", () => {
+			const previousPackageName = `${packageObject.name}-does-not-exist`;
+			assert.throws(() => {
+				utils.ensureDevDependencyExists(packageObject, previousPackageName);
+			});
+		});
+	});
+
+	describe("Test getPreviousPackageJsonPath", () => {
```

Comment:
I would recommend adding a failure mode test case as well. See here for an example pattern: https://github.com/microsoft/FluidFramework/blob/main/packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts#L127